src/main/ruby/truffleruby/core/comparable.rb

```diff
@@ -79,7 +79,20 @@ def between?(min, max)
     true
   end
 
-  def clamp(min, max)
+  def clamp(*args)
+    case args.count
+    when 1
+      range = args[0]
+      raise TypeError, "wrong argument type #{range.class.name} (expected Range)" unless range.kind_of?(Range)
+      raise ArgumentError, 'cannot clamp with an exclusive range' if range.exclude_end?
+
+      min, max = range.begin, range.end
+    when 2
+      min, max = args[0], args[1]
+    else
+      raise ArgumentError, "wrong number of arguments (given #{args.count}, expected 1..2)"
```

Comment:
I much prefer to naturally check the arguments and have more useful parameters for the method, I'll switch to:
```ruby
  def clamp(min, max = undefined)
```
That avoids this branch, which notably duplicates the error message, makes it nicer for `Method#parameters` and I think it's easier to read.

Reply:
Yeah I'd agree with that change. Do you want to make it? It's already approved.

Reply:
Yeah I did it (I wrote `I'll switch to` above).